src/LodeRunner.Data/Services/BaseService.cs

```diff
@@ -34,10 +48,9 @@ public BaseService(ICosmosDBRepository cosmosDBRepository)
         /// <summary>
         /// Gets the specified identifier.
         /// </summary>
-        /// <typeparam name="TEntity">The type of the entity.</typeparam>
         /// <param name="id">The identifier.</param>
         /// <returns>The corresponding entity.</returns>
-        public virtual async Task<TEntity> Get<TEntity>(string id)
+        public virtual async Task<TEntity> Get(string id)
         {
             EntityType entityType = typeof(TEntity).Name.As<EntityType>();
```

Comment:
```csharp
EntityType entityType = typeof(TEntity).Name.As<EntityType>();
``` 
Move this into the constructor and determine the EntityType once as the object will have the same EntityType no matter which method is called on the instantiated object.

Reply:
@gortegaMS check with @jofultz

Reply:
Refactror EntityType into a private member